### PR TITLE
fix: merge oclif, up deps, and set new command folder

### LINKED ISSUE #34: "Buffer() is deprecated"
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------
A bug

What is the current behavior?
-----------------------------
Running `sfdx plugins:generate` issues the following warning after running `yard add v1.0.1`:

```
(node:12066) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```

What is the expected behavior?
------------------------------
Expected behavior: No warning

sfdx-cli: 6.34.0-6fec0d751a (darwin-x64) node-v8.9.4
OS version: macOS 10.14 Mojave

### LINKED ISSUE #37: Error thrown when running tests
Reporting a *bug*
-----------------------------------------------------
Error "INVALID_SESSION_ID: Invalid Session ID found in SessionHeader: Illegal Session" thrown when running tests of a command which includes a method call "conn.metadata.describe".

What is the current behavior?
-----------------------------
I'm trying to test a command which includes the following code:
```
    const conn = this.org.getConnection();
    const result = await conn.metadata.describe(await this.org.retrieveMaxApiVersion());
    this.ux.log(result);
```
with the following test:
```
describe('hello:metadata', () => {
  test
    .withOrg({ username: 'test@org.com' }, true)
    .withConnectionRequest(request => Promise.resolve({ metadataObjects: [ { directoryName: 'objects', inFolder: false, metaFile: false, suffix: 'object', xmlName: 'CustomObject' }] }))
    .stdout()
    .command(['hello:metadata', '--targetusername', 'test@org.com'])
    .it('runs hello:metadata --targetusername test@org.com', ctx => {
      expect(ctx.stdout).to.contain('objects');
    });
});
```
But I'm always receiving the following error: "INVALID_SESSION_ID: Invalid Session ID found in SessionHeader: Illegal Session"

What is the expected behavior?
------------------------------
$ sfdx --version
sfdx-cli/6.35.1-8b7320f7c2 (darwin-x64) node-v8.9.4

### PR DESCRIPTION
**Pull request by amphro**
Updated deps for #55 
I get clean tests now, so should resolve #37 
I don't think this was a problem before this change, but updated deps so should also resolve #34

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
         "description": "add/remove/create CLI plug-ins"
       }
     },
-    "commands": "./lib/commands",
+    "commands": "./lib/sfdxcommands",
```
Comment (amphro) [inline package.json:71]: @rbergman this will make it so we don't have the hidden oclif commands we don't want on suggestions.

Comment (rbergman) [reply package.json:71]: nice!

File: src/generators/app.ts [hunk 2]
```diff
@@ -454,18 +462,17 @@ class App extends Generator {
...
+        '@salesforce/command@^1.4.1',
+        '@salesforce/core@^1.3.2'
       )
       devDependencies.push(
         '@oclif/dev-cli@^1',
         '@oclif/plugin-help@^2',
         'globby@^8',
-        '@salesforce/dev-config@^1.1.4',
-        'sinon@5',
+        '@salesforce/dev-config@1.4.1'
```
Comment (rbergman) [inline src/generators/app.ts:475]: Latest is 1.4.4, any reason to not use that one?

File: src/generators/app.ts [hunk 3]
```diff
@@ -491,25 +498,28 @@ class App extends Generator {
...
-        'tslib@1',
+        'tslib@^1',
       )
       devDependencies.push(
-        '@types/chai@^4',
-        '@types/mocha@^5',
         '@types/node@^10',
-        'typescript@^3.0',
-        'ts-node@^7',
+        'typescript@^3.3',
```
Comment (rbergman) [inline src/generators/app.ts]: I'm wary about careting ts -- they sometimes introduce breaking changes in minor releases.  Maybe ~ would be better here.

**Review by rbergman — APPROVED** [review]

File: src/generators/app.ts (hunk 3 above)
Comment (amphro) [reply src/generators/app.ts]: Sure

**Pull request merged**